v-next/hardhat/src/internal/cli/telemetry/sentry/anonymizer.ts

```diff
@@ -166,38 +215,47 @@ export class Anonymizer {
         continue;
       }
 
-      if (this.#errorRaisedByPackageToIgnore(frame.filename)) {
+      // We don't report errors from the Hardhat.config file
+      if (
+        this.#configPath !== undefined &&
+        this.#configPath.includes(frame.filename)
+      ) {
         return false;
       }
 
-      // we stop after finding either a hardhat file or a file from the user's
-      // project
-      if (this.#isHardhatFile(frame.filename)) {
-        return true;
-      }
-
-      if (frame.filename === ANONYMIZED_FILE) {
+      // We don't report errors from the ignored package list e.g. `ethers`
+      if (this.#errorRaisedByPackageToIgnore(frame.filename)) {
         return false;
       }
 
+      // We don't report errors originating from non-Hardhat packages
       if (
-        this.#configPath !== undefined &&
-        this.#configPath.includes(frame.filename)
+        this.#isPackageFile(frame.filename) &&
+        !this.#isHardhatFile(frame.filename)
       ) {
         return false;
       }
+
+      // TODO: For the moment we are loose and treat as Hardhat errors
+      // anything not caught as by the explicit opt-outs above.
+      // Right now this will include files from the user's project
+      // (i.e. js/ts files not under `./node_modules`)
+      // We will tighten these up based on Sentry examples.
+      return true;
```

Comment:
Right now we are not filtering out errors where the originating file is outside of `node_modules` i.e. a script or ignition module or such.